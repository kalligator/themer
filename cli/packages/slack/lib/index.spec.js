const { render, renderInstructions } = require('./index'),
  { colors } = require('../../colors-default');

describe('render', () => {
  const promisedFiles = Promise.all(render(colors));

  it('should render the expected number of files', async () => {
    const files = await promisedFiles;
    expect(files.length).toBe(2);
  });

  it('should render valid Slack theme strings', async () => {
    const files = await promisedFiles;
    files.forEach((file) => {
      const contents = file.contents.toString('utf8');
      expect(/\,\,/.test(contents)).toBe(false);
      expect(/\s/.test(contents)).toBe(false);
    });
  });
});

describe('renderInstructions', () => {
  it('should provide installation instructions', async () => {
    const files = await Promise.all(render(colors));
    const instructions = renderInstructions(files.map(({ name }) => name));
    expect(instructions).toMatchSnapshot();
  });
});
