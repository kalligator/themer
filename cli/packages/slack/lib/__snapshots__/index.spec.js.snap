// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderInstructions should provide installation instructions 1`] = `
"
Copy the contents of \`themer-slack-dark.txt\` or \`themer-slack-light.txt\` and paste into the custom theme input in Slack's preferences.
"
`;
